pkg/operation/botanist/component/shared/new.go

```diff
@@ -0,0 +1,87 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"time"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/resourcemanager"
+	"github.com/gardener/gardener/pkg/utils/images"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+
+	"github.com/Masterminds/semver"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/component-base/version"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewGardenerResourceManager instantiates a new `gardener-resource-manager` component.
+func NewGardenerResourceManager(
+	c client.Client,
+	gardenNamespaceName string,
+	runtimeVersion *semver.Version,
+	imageVector imagevector.ImageVector,
+	secretsManager secretsmanager.Interface,
+	logLevel, logFormat string,
+	secretNameServerCA string,
+	defaultSeccompProfileEnabled bool,
+	zones []string,
+) (
+	component.DeployWaiter,
+	error,
+) {
+	image, err := imageVector.FindImage(images.ImageNameGardenerResourceManager)
+	if err != nil {
+		return nil, err
+	}
+
+	repository, tag := image.String(), version.Get().GitVersion
+	if image.Tag != nil {
+		repository, tag = image.Repository, *image.Tag
+	}
+	image = &imagevector.Image{Repository: repository, Tag: &tag}
+
+	return resourcemanager.New(c, gardenNamespaceName, secretsManager, resourcemanager.Values{
+		ConcurrentSyncs:                      pointer.Int(20),
+		DefaultSeccompProfileEnabled:         defaultSeccompProfileEnabled,
+		HealthSyncPeriod:                     &metav1.Duration{Duration: time.Minute},
+		Image:                                image.String(),
+		LogLevel:                             logLevel,
+		LogFormat:                            logFormat,
+		MaxConcurrentRootCAPublisherWorkers:  pointer.Int(5),
+		MaxConcurrentTokenInvalidatorWorkers: pointer.Int(5),
+		// TODO(timuthy): Remove PodTopologySpreadConstraints webhook once for all seeds the
+		//  MatchLabelKeysInPodTopologySpread feature gate is beta and enabled by default (probably 1.26+).
+		PodTopologySpreadConstraintsEnabled: true,
+		Replicas:                            pointer.Int32(2),
+		ResourceClass:                       pointer.String(v1beta1constants.SeedResourceManagerClass),
```

Comment:
> but changing the class later will be too much effort. 

Probably not, as discussed on the phone yesterday. We can work on this at a later point in time when introducing a garden care controller. For now it's not a problem to just re-use this, but we might want to indeed change it later (also, because `ManagedResource`s are labeled with `type=seed-system-component`) by default. Let's work on this in a dedicated workstream/with dedicated focus.